src/Maths/Silk.NET.Intrinsics/Avx/AvxRegister.Float.cs

```diff
@@ -0,0 +1,330 @@
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public WorkUnit<float> Transform4(WorkUnit<float> value, WorkUnit<float> quaternion)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public WorkUnit<float> Abs(WorkUnit<float> vector)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public WorkUnit<float> HorizontalAdd(WorkUnit<float> left, WorkUnit<float> right)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public WorkUnit<float> Shuffle(WorkUnit<float> value, byte control)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public WorkUnit<float> X(WorkUnit<float> vector)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public WorkUnit<float> Y(WorkUnit<float> vector)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public WorkUnit<float> Z(WorkUnit<float> vector)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public WorkUnit<float> W(WorkUnit<float> vector)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public WorkUnit<float> Sin(WorkUnit<float> vector)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public WorkUnit<float> SinApprox(WorkUnit<float> vector)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public WorkUnit<float> Cos(WorkUnit<float> vector)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public WorkUnit<float> CosApprox(WorkUnit<float> vector)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public WorkUnit<float> Tan(WorkUnit<float> vector)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public WorkUnit<float> TanApprox(WorkUnit<float> vector)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public void SinCos(WorkUnit<float> vector, out WorkUnit<float> sin, out WorkUnit<float> cos)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public WorkUnit<float> Shuffle(WorkUnit<float> left, WorkUnit<float> right, byte control)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public WorkUnit<float> MultiplyAddFused(WorkUnit<float> x, WorkUnit<float> y, WorkUnit<float> z)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public WorkUnit<float> MultiplyAddFast(WorkUnit<float> x, WorkUnit<float> y, WorkUnit<float> z)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public WorkUnit<float> NegateMultiplyAddFused(WorkUnit<float> x, WorkUnit<float> y, WorkUnit<float> z)
```

Comment:
Why is an explicit `NegateMultiplyAdd` needed? Seems like an optimization the JIT should (and does) do...

Reply:
This will use the FMA register if applicable, otherwise it will use your everyday AVX register.

Reply:
Sure, but I'm not sure that clarifies why it is needed?

I can't think of an optimization you can do knowing that it is `-(a * b) + c` vs `(a * b) + c`, in which case you should be able to just have `MultiplyFusedAdd` and let the JIT optimize `MultiplyFusedAdd(Negate(a), b, c)` when FMA is available and just do your normal math, including the negation, when it isn't

Reply:
Very good point, cc @sunkin351